File: src/main/java/dev/robothanzo/werewolf/config/SessionConfig.java
package dev.robothanzo.werewolf.config;

import org.mongodb.spring.session.config.annotation.web.http.EnableMongoHttpSession;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.session.web.http.CookieSerializer;
import org.springframework.session.web.http.DefaultCookieSerializer;

@Configuration
@EnableMongoHttpSession(collectionName = "http_sessions")
public class SessionConfig {
    
    @Bean
    public CookieSerializer cookieSerializer() {
        DefaultCookieSerializer serializer = new DefaultCookieSerializer();
        serializer.setSameSite("Lax");
        serializer.setUseHttpOnlyCookie(true);
        serializer.setUseSecureCookie(!"http://localhost:5173".equals(System.getenv().getOrDefault("DASHBOARD_URL", "http://localhost:5173")));
        return serializer;
    }
}

File: src/main/kotlin/dev/robothanzo/werewolf/config/SessionConfig.kt
package dev.robothanzo.werewolf.config

import dev.robothanzo.werewolf.database.documents.Session
import org.mongodb.spring.session.config.annotation.web.http.EnableMongoHttpSession
import org.springframework.context.annotation.Configuration

@Configuration
@EnableMongoHttpSession(collectionName = "http_sessions")
class SessionConfig {
    @org.springframework.context.annotation.Bean
    fun cookieSerializer(): org.springframework.session.web.http.CookieSerializer {
        val serializer = org.springframework.session.web.http.DefaultCookieSerializer()
        serializer.setSameSite("Lax")
        serializer.setUseHttpOnlyCookie(true)

        // Use secure cookies for HTTPS environments (production)
        val dashboardUrl = Session.DASHBOARD_BASE_URL
        val isSecureEnvironment = dashboardUrl.startsWith("https://")
        serializer.setUseSecureCookie(isSecureEnvironment)

        return serializer
    }
}
